src/azure-cli/azure/cli/command_modules/storage/operations/file.py

```diff
@@ -208,15 +208,42 @@ def _upload_action(src, dst2):
     return list(_upload_action(src, dst) for src, dst in source_files)
 
 
-def storage_file_download_batch(cmd, client, source, destination, pattern=None, dryrun=False, validate_content=False,
-                                max_connections=1, progress_callback=None, snapshot=None):
+def download_file(client, destination_path=None, timeout=None, max_connections=2, open_mode='wb', **kwargs):
+    from azure.cli.command_modules.storage.util import mkdir_p
+    destination_folder = os.path.dirname(destination_path) if destination_path else ""
+    if destination_folder and not os.path.exists(destination_folder):
+        mkdir_p(destination_folder)
+
+    if not destination_folder or os.path.isdir(destination_path):
+        file = client.get_file_properties(timeout=timeout)
+        file_name = file.name.split("/")[-1]
+        destination_path = os.path.join(destination_path, file_name) \
+            if destination_path else file_name
+
+    kwargs['raw_response_hook'] = kwargs.pop("progress_callback", None)
+
+    with open(destination_path, open_mode) as stream:
+        start_range = kwargs.pop('start_range', None)
+        end_range = kwargs.pop('end_range', None)
+        length = None
+        if start_range is not None and end_range is not None:
+            length = end_range - start_range + 1
+        download = client.download_file(offset=start_range, length=length, timeout=timeout,
+                                        max_concurrency=max_connections, **kwargs)
+        download_content = download.readall()
+        stream.write(download_content)
```

Comment:
`download.readinto(stream)` would be better because `download.readall()` may cause memory overhead when `download_content` is quite large